### PR TITLE
Change migrations logic for rcce tag

### PR DESCRIPTION
**Pull request by k9845**
Addresses 
 - https://github.com/IFRCGo/go-frontend/issues/1970

TODO: Remove test migrations and `RCCE` from SectorTags
NOTE: This is a fix for failed migrations

## Changes
- Added test for migrations
- Change in migrations logic

## Checklist 
Things that should succeed before merging.

- [x] Updated/ran unit tests
- [ ] Updated CHANGELOG.md

## Release

If there is a version update, make sure to tag the repository with the latest version.

### CODE REVIEW

File: deployments/models.py [hunk 1]
```diff
@@ -290,7 +290,7 @@ class SectorTags(IntEnum):
     HEALTH_PUBLIC = 4
     HEALTH_CLINICAL = 12
     COVID_19 = 13
-    RRCE = 14
+    RCCE = 14
```
Comment (thenav56) [inline deployments/models.py:293]: We don't need this now.

File: deployments/test_migrations.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        project_1 = ProjectFactory.create(secondary_sectors=[SectorTags.RCCE])
+        project_2 = ProjectFactory.create(secondary_sectors=[SectorTags.RCCE, SectorTags.MIGRATION, SectorTags.WASH])
+
+        # Apply the migration logic
+        migration_file._remove_rcce_tag(Project)
+
+        assert set(Project.objects.filter(id=project_1.id).values_list('secondary_sectors', flat=True)[0])\
+            == set([SectorTags.HEALTH_PUBLIC, SectorTags.CEA])
+        assert set(Project.objects.filter(id=project_2.id).values_list('secondary_sectors', flat=True)[0])\
+            == set([SectorTags.HEALTH_PUBLIC, SectorTags.CEA, SectorTags.MIGRATION, SectorTags.WASH])
```
Comment (thenav56) [inline deployments/test_migrations.py:27]: Let's update the test case with more conditions. Also, add a TODO: and a NOTE in PR as well.

```suggestion
   # TODO: This is a temporary test for PR only, after merging we will remove this (and also RCCE from SectorTags).
    def test_remove_rcce_tag(self):
        migration_file = importlib.import_module('deployments.migrations.0039_auto_20210823_1159')

        project_0 = ProjectFactory.create(secondary_sectors=[])  # Without any RCCE tag
        project_1 = ProjectFactory.create(secondary_sectors=[SectorTags.RCCE]) # Without just RCCE tag
        project_2 = ProjectFactory.create(   # Wit} just RCCE tag + other tags
            secondary_sectors=[SectorTags.RCCE, SectorTags.MIGRATION, SectorTags.WASH]
        )
        project_3 = ProjectFactory.create(  # With other tags
            secondary_sectors=[SectorTags.MIGRATION, SectorTags.WASH]
        )
        project_4 = ProjectFactory.create(  # With RCCE + CEA and HEALTH_PUBLIC
            secondary_sectors=[SectorTags.RCCE, SectorTags.MIGRATION, SectorTags.WASH]
        )
        project_5 = ProjectFactory.create(  # With RCCE + CEA
            secondary_sectors=[SectorTags.RCCE, SectorTags.CEA]
        )

        # Apply the migration logic
        migration_file._remove_rcce_tag(Project)

        project_0.refresh_from_db()
        project_1.refresh_from_db()
        project_2.refresh_from_db()
        project_3.refresh_from_db()
        project_4.refresh_from_db()
        project_5.refresh_from_db()

        assert set(project_0.secondary_sectors) == set([])
        assert set(project_1.secondary_sectors) == set([SectorTags.HEALTH_PUBLIC, SectorTags.CEA])
        assert set(project_2.secondary_sectors) == set([SectorTags.HEALTH_PUBLIC, SectorTags.CEA, SectorTags.MIGRATION, SectorTags.WASH])
        assert set(project_3.secondary_sectors) == set([SectorTags.MIGRATION, SectorTags.WASH])
        assert set(project_4.secondary_sectors) == set([SectorTags.HEALTH_PUBLIC, SectorTags.CEA])
        assert set(project_5.secondary_sectors) == set([SectorTags.HEALTH_PUBLIC, SectorTags.CEA])
```

**Review by thenav56 — CHANGES_REQUESTED** [review]

**Review by thenav56 — APPROVED** [review]

**Comment by thenav56** [comment]
@szabozoltan69 Could you merge this?

This PR basically has a test case for migration from RCCE -> Health + CEE.
We will send another PR where we will remove RCCE and the test case (which will fail without RCCE but the migration will work without RCCE)

**Review by szabozoltan69 — APPROVED** [review]
Great stuff! Even testing the migration...

**Comment by szabozoltan69** [comment]
Merged to develop, deployed to staging

**Pull request merged**